app.go

```diff
@@ -579,9 +579,20 @@ func (a *Application) DeliverTx(txBytes []byte) abci.ResponseDeliverTx {
 		deliverTxLatency.With(lvs...).Observe(time.Since(begin).Seconds())
 	}(time.Now())
 
+	state := NewStoreState(
+		context.Background(),
+		a.Store,
+		a.curBlockHeader,
+		a.curBlockHash,
+		a.GetValidatorSet,
+	)
+
 	r, err := a.processTx(txBytes, false)
 	if err != nil {
 		log.Error(fmt.Sprintf("DeliverTx: %s", err.Error()))
+		if state.FeatureEnabled(EvmTxReceiptsVersion2_1Feature, false) {
+			return abci.ResponseDeliverTx{Code: 1, Data: r.Data, Log: err.Error()}
+		}
```

Comment:
Can you explain why this changed?

Reply:
For success tx, the application returns receipt tx hash back to Tendermint, so the receipt tx hash is used as a transaction hash but for failed tx, the application does not return receipt tx hash back to Tendermint, so it creates its own tx hash. So the tx hash on the receipt and transaction tx hash are different. As a result, we cannot query receipt using Tendermint tx hash.

Reply:
Receipt Tx hash is created here and returned to Tendermint through ResponseDeliverTx https://github.com/loomnetwork/loomchain/blob/445a4053cf53ac1582c7cb5dd5dbd688b2a048a0/receipts/leveldb/leveldb.go#L55

Reply:
Actually, you can query for a receipt using `Tendermint transaction hash` `(= types.Tx(txBytes).Hash())`. 
The receipts code is doubled up so you can get the receipt from both `tendermint` and `loom` tx hash. This was done so that you could get receipts from failed txs and the rtv from `eth_sendRawTransaction`. 

What we would probably want to do is replace the loom-tx hash with one of
* Use the tenderment tx hash throughout which would allow us to greatly simplify the tx-receipt code.
* Use  ethereum tx hash. Which would mean having to continue using two tx hashes and could keep the code much as it is.

Reply:
@Sriep are you referencing to this PR https://github.com/loomnetwork/loomchain/pull/1401? Right now it seems that we can only query receipt using loom tx hash not tm tx hash.